plugins/MidiImport/MidiImport.cpp

```diff
@@ -374,19 +374,22 @@ bool MidiImport::readSMF( TrackContainer* tc )
 			if( evt->chan == -1 )
 			{
 				bool handled = false;
-				if( evt->is_update() )
+                if( evt->is_update() )
 				{
 					QString attr = evt->get_attribute();
-					if( attr == "tracknames" ) {
+                    if( attr == "tracknames" && evt->get_update_type() == 'a' ) {
```

Comment:
Yeah, better to be safe than sorry. I wonder where a trackname event will hold the track name if it doesn't have an atom to read, though.